docs/developer/documentation.md

```diff
@@ -0,0 +1,163 @@
+# NGINX Gateway Fabric Docs
```

Comment:
would it make sense to add a note that hugo commands must be run from the `/site` directory?
I first tried to run it from the root directory :)

Reply:
```
# NGINX Gateway Fabric Docs

The `/site` directory contains the user documentation for NGINX Gateway Fabric and the requirements for linting, building, and publishing the docs. Run all the `hugo` commands below from this directory.
```